packages/shared/src/lib/core/nfts/actions/addOrUpdateNftInAllAccountNfts.ts

```diff
@@ -17,3 +17,19 @@ export function addOrUpdateNftInAllAccountNfts(accountIndex: number, ...newNfts:
         return state
     })
 }
+
+export function addOrUpdateNftInAllAccountNftsForAllAccounts(...newNfts: INft[]): void {
+    allAccountNfts.update((state) => {
+        for (const accountNfts of Object.values(state)) {
+            for (const newNft of newNfts) {
+                const nft = accountNfts.find((_nft) => _nft.id === newNft.id)
+                if (nft) {
+                    Object.assign(nft, newNft)
+                } else {
+                    accountNfts.push(newNft)
+                }
+            }
```

Comment:
This logic has been duplicated in the function above. This function has a confusing name as well. Maybe `addOrUpdateAllAccountNfts` would do the job?